games/xworld/maps/xworld_env.py

```diff
@@ -193,17 +203,26 @@ def check_or_get_value(valid_value_set, is_continuous=False):
             assert property in entity.__dict__.keys() and property in default_dict.keys(), \
                 "invalid property name: %s is provided" % property
             value = context_dict[property]
-            if value == False: # should only be used for properties with values
+            if value is False: # should only be used for properties with values
                 assert entity.__dict__[property], "no existing value for property %s" % property
                 continue
+            elif value is None and entity.__dict__[property] is None:
+                #reinstantiate if property value doesn't exist
+                value = True
+            elif value is None:
+                continue
```

Comment:
Why do you need True, False, and None for value? The logic here is complex. I believe a cleaner design exists.

Reply:
These options  are included for handling existing use cases. May re-design in the future.